### PR TITLE
current-repodata plugin + update conda-suggest and repodata-patching plugin

### PR DESCRIPTION
**Pull request by madhur-tandon**
1. A new plugin `quetz_current_repodata` without an `api.py` i.e. no endpoint. Thus, it only contains the `hook`
2. The `index.html` is generated after all such `hooks` are called and the `files` parameter is updated (defined in `indexing.py`). This is different from using `pkgstore.list_files` since that would also give us data for multiple `subdirs` + would also list `packages` -- which is all extra information.
3. No tests are written for the new plugin `quetz_current_repodata` since there is no API endpoint.
4. The `quetz_conda_suggest` plugin is updated so as to use `pkgstore.add_file()` to create the `.map` file. It also updates the `index.html` so that `.map` appears in it.

### CODE REVIEW

File: plugins/quetz_current_repodata/quetz_current_repodata/main.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import hashlib
+import json
+from datetime import datetime, timezone
+
+from conda_build.index import _build_current_repodata
+from fastapi import APIRouter
+
+import quetz
+
+router = APIRouter()
```
Comment (wolfv) [inline plugins/quetz_current_repodata/quetz_current_repodata/main.py:11]: unused?

Comment (madhur-tandon) [reply plugins/quetz_current_repodata/quetz_current_repodata/main.py:11]: Yeah, I removed it before but then the below `register_router` hook needs it. OR can I remove the hook below as well?

File: plugins/quetz_current_repodata/quetz_current_repodata/main.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+        updated_files.append((fname, raw_current_repodata))
+        updated_files.append((fname + ".bz2", compressed_current_repodata))
+
+        update_index(pkgstore, updated_files, channel_name, subdir, files)
+
+
+def update_index(pkgstore, updated_files, channel_name, subdir, files):
+    for fname, data in updated_files:
+        md5 = hashlib.md5()
```
Comment (wolfv) [inline plugins/quetz_current_repodata/quetz_current_repodata/main.py:50]: maybe we can factor this md5 & sha256 computation, and adding the file to `files` into a function? I guess we're doing this from a couple of places now.

Comment (madhur-tandon) [reply plugins/quetz_current_repodata/quetz_current_repodata/main.py]: Done!

File: plugins/quetz_current_repodata/quetz_current_repodata/main.py (hunk 1 above)
Comment (wolfv) [reply plugins/quetz_current_repodata/quetz_current_repodata/main.py:10]: I think so :)

**Comment by wolfv** [comment]
nice!

File: plugins/quetz_current_repodata/quetz_current_repodata/main.py (hunk 1 above)
Comment (madhur-tandon) [reply plugins/quetz_current_repodata/quetz_current_repodata/main.py]: Done!

**Comment by wolfv** [comment]
hey, congrats! Looks good to me :+1:

Do you want to squash the commits a little bit? Gonna merge right after.

**Comment by madhur-tandon** [comment]
Do you need README.md + tests for the `current-repodata` plugin?

**Comment by wolfv** [comment]
ideally, yes :)

**Pull request merged**